src/ansys/pytwin/evaluate/twin_model.py

```diff
@@ -975,6 +1333,114 @@ def save_state(self):
+        named_selection: str (optional)
+            Named selection on which the snasphot has to be generated
+
+        Raises
+        ------
+        TwinModelError:
+            If rom_name is not included in the Twin's list of TBROM
+            If name_selection is not included in the TBROM's list of Named Selections
+
+        Examples
+        --------
+        >>> from pytwin import TwinModel
+        >>> # Instantiate a twin model, initialize it, and evaluate it step by step until you want to save its state
+        >>> model1 = TwinModel('model.twin')
+        >>> model1.initialize_evaluation()
+        >>> romname = model1.tbrom_names[0]
+        >>> nslist = model1.get_rom_nslist(romname)
+        >>> fieldresults = model1.snapshot_generation(romname, False, nslist[0])
+        """
+        self._log_key = "SnapshotGeneration"
+
+        try:
+            if named_selection is not None:
+                if self._check_tbrom_snapshot_generation_args(rom_name, named_selection):
+                    output_file = (
+                        self._tbroms[rom_name].outputfieldname
+                        + "_"
+                        + named_selection
+                        + "_"
+                        + "{:.6f}".format(self.evaluation_time)
+                        + ".bin"
+                    )
+                    output_file_path = os.path.join(self._tbroms[rom_name]._outputfilespath, output_file)
+                    return self._tbroms[rom_name].snapshot_generation(on_disk, output_file_path, named_selection)
+            else:
+                if self._check_tbrom_snapshot_generation_args(rom_name):
+                    output_file = (
+                        self._tbroms[rom_name].outputfieldname + "_" + "{:.6f}".format(self.evaluation_time) + ".bin"
+                    )
+                    output_file_path = os.path.join(self._tbroms[rom_name]._outputfilespath, output_file)
+                    return self._tbroms[rom_name].snapshot_generation(on_disk, output_file_path, named_selection)
+
+        except Exception as e:
+            msg = f"Something went wrong while generating the snapshot:"
+            msg += f"\n{str(e)}."
+            self._raise_error(msg)
+
+    def points_generation(self, rom_name: str, on_disk: bool = True, named_selection: str = None):
+        """
+        Generate a points file either in memory or on disk, for the full field or a specific part. It returns the field
+        data as an array if in memory, or the path of the snapshot written on disk.
+
+        Parameters
+        ----------
+        rom_name: str
+            TBROM name part of the Twin for which a point file has to be generated
+        on_disk: bool
+            Whether the point file is saved on disk (True) or returned in memory (False)
+        named_selection: str (optional)
+            Named selection on which the point file has to be generated
```

Comment:
```suggestion
        rom_name : str
            TBROM name part of the twin to generate the point file on.
        on_disk : bool
            Whether the point file is saved on disk (True) or returned in memory (False).
        named_selection : str, optional
            Named selection to generate the point file on.
```
Again--changes here and earlier may be incorrect--"name part" still seems odd. Please make the same changes in both parts.